### PR TITLE
Bug 1826051: Support Cluster Trigger Binding

### PR DESCRIPTION
**Pull request by andrewballantyne**
**Fixes**: 
https://issues.redhat.com/browse/ODC-3638

**Analysis / Root cause**: 
With the addition of 0.11 Pipeline Operator, the Trigger code for 4.5 no longer works (group change); also supports the Operator orientated ClusterTriggerBinding model.

**Solution Description**: 
Support ClusterTriggerBinding and fix up the code to use it.

**Screen shots / Gifs for design review**: 
image

cc @openshift/team-devconsole-ux 

**Unit test coverage report**: 

Had to cut some bad tests out, can't be deep diving otherwise it makes brittle tests like this one.

**Test setup:**

* Install `canary` on the Community OpenShift Pipelines Operator (which should give you 0.11)
* Head over and create a Pipeline (oc create if you want)
* Add a Trigger to the Pipeline (note that the Operator installed some ClusterTriggerBindings, use one of those)
* Head over to the EventListener created (Pipeline Details => TriggerBinding Details => EventListener Details)
** You'll see it shows the CTB (ClusterTriggerBinding) resource now

**Browser conformance**: 
- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by andrewballantyne** [comment]
/assign @karthikjeeyar 

Please review this PR as it changed some of your code. You nixed ClusterTasks in your Pipeline Tasks ResourceLink change.

File: frontend/packages/dev-console/src/components/pipelines/resource-overview/__tests__/DynamicResourceLinkList.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { shallow, ShallowWrapper } from 'enzyme';
+import { TaskModel, ClusterTaskModel } from '../../../../models';
+import DynamicResourceLinkList from '../DynamicResourceLinkList';
+import { ResourceLink } from '@console/internal/components/utils';
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/components/pipelines/resource-overview/__tests__/DynamicResourceLinkList.spec.tsx:5]: import statement needs sorting.

**Comment by karthikjeeyar** [comment]
@andrewballantyne  Since we are reusing the TriggerBindingPage/TriggerBindingDetails component in ClusterTriggerBinding details page, the title has to be changed based on the model.

image

**Comment by andrewballantyne** [comment]
@karthikjeeyar Please re-review, updated to fix the title.

**Comment by karthikjeeyar** [comment]
Verified locally, supports cluster task binding as expected.
/lgtm

**Comment by andrewballantyne** [comment]
/hold

I saw an issue that needs to be addressed, will push momentarily.

**Comment by andrewballantyne** [comment]
/hold cancel

@karthikjeeyar okay, I think I got the last reference where it just assumed Type for the TriggerBinding. We should be good to go *now* 😛 One more review please.

**Comment by andrewballantyne** [comment]
/hold cancel

@karthikjeeyar okay, I think I got the last reference where it just assumed Type for the TriggerBinding. We should be good to go *now* 😛 One more review please.

**Comment by andrewballantyne** [comment]
While testing this out, I noticed their label that allowed us to show what triggered the PipelineRun changed as well... so I pushed another tiny change.

**Pull request merged**